### PR TITLE
quiche: roll QUICHE to 9711a9e74b43 and BoringSSL to 65e0aad1b721

### PR DESCRIPTION
**Pull request by bencebeky**
Signed-off-by: Bence Béky <bnc@google.com>

Description: roll QUICHE and BoringSSL external dependencies to more recent versions and make necessary modifications.

Risk Level: Low.

Testing: No new tests.  Both external dependencies are covered by extensive unit tests.  No new functionality is added to Envoy, therefore existing integration tests should be sufficient.

Docs Changes: n/a.

Release Notes: n/a.

### CODE REVIEW

File: bazel/external/quiche.BUILD [hunk 1]
```diff
@@ -2141,8 +2144,8 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "quic_core_http_spdy_stream_body_buffer_lib",
```
Comment (danzh2010) [inline bazel/external/quiche.BUILD]: Can you also rename this library to reflect the file name?

Comment (bencebeky) [reply bazel/external/quiche.BUILD]: Good catch, thank you.

File: bazel/external/quiche.BUILD [hunk 2]
```diff
@@ -3408,3 +3399,37 @@ envoy_cc_test(
         ":quic_platform_test_mem_slice_vector_lib",
     ],
 )
+
+envoy_cc_library(
```
Comment (danzh2010) [inline bazel/external/quiche.BUILD]: nits: mind placing the new build target in alphabetical order?

Comment (bencebeky) [reply bazel/external/quiche.BUILD]: Of course, as much as it's possible (it's not exactly sorted currently, but close).

**Comment by bencebeky** [comment]
/assign mattklein123

This change is ready for review.

**Comment by bencebeky** [comment]
/unassign danzh2010
/assign alyssawilk

Unassigning danzh2010 who has already finished reviewing.
Assigning other code owner as this change is ready for review from either owners.  Thank you.

**Pull request merged**